### PR TITLE
client/tailscale: update Client API a bit

### PR DESCRIPTION
**Pull request by bradfitz**

### CODE REVIEW

**Comment by bradfitz** [comment]
I have the corresponding change on the other side just about ready too. I won't submit this one until that one's also out & approved.

File: client/tailscale/tailscale.go [hunk 1]
```diff
@@ -30,45 +30,90 @@ var I_Acknowledge_This_API_Is_Unstable = false
...
+// AuthMethod is the interface for API authentication methods.
+//
+// Most users will use AuthKey.
+type AuthMethod interface {
+	modifyRequest(req *http.Request)
+}
+
+// AuthKey is an AuthMethod for NewClient that authenticates requests
+// using an authkey.
+type AuthKey string
```
Comment (maisem) [inline client/tailscale/tailscale.go:82]: APIKey
AuthKey is used specifically for registration of nodes

File: client/tailscale/tailscale.go [hunk 2]
```diff
@@ -9,7 +9,7 @@
 // Tailscale control plane API.
 //
 // Warning: this package is in development and makes no API compatibility
-// promises as of 2022-04-29. It is subject to change at any time.
+// promises as of 2022-04-29. It is subject to change at any time.q
```
Comment (maisem) [inline client/tailscale/tailscale.go:12]: Q?

File: client/tailscale/acl.go [hunk 3]
```diff
@@ -436,14 +436,14 @@ func (c *Client) ValidateACLJSON(ctx context.Context, source, dest string) (test
...
 
-	path := fmt.Sprintf("%s/api/v2/tailnet/%s/acl/validate", c.BaseURL, c.Tailnet)
+	path := fmt.Sprintf("%s/api/v2/tailnet/%s/acl/validate", c.baseURL(), c.tailnet)
 	req, err := http.NewRequestWithContext(ctx, "POST", path, bytes.NewBuffer(postData))
 	if err != nil {
 		return nil, err
 	}
 
 	req.Header.Set("Content-Type", "application/json")
-	req.SetBasicAuth(c.APIKey, "")
```
Comment (maisem) [inline client/tailscale/acl.go:446]: It used to be APIKey too

**Review by maisem — APPROVED** [review]

File: client/tailscale/tailscale.go (hunk 1 above)
Comment (bradfitz) [reply client/tailscale/tailscale.go:82]: braino

**Pull request merged**